client/ayon_core/pipeline/publish/abstract_collect_render.py

```diff
@@ -215,7 +219,16 @@ def process(self, context):
             render_instance_dict = attr.asdict(render_instance)
 
             instance = context.create_instance(render_instance.name)
-            instance.data["label"] = render_instance.label
+            if render_instance.source_instance is not None:
+                # remove the new instance, because we want to insert it
```

Comment:
Maybe `replace_instance_in_context` could care about this too?

```python
if src_instance in context:
    context.remove(src_instance)

if dst_instance not in context:
    context.append(dst_instance)
```

Reply:
> Maybe `replace_instance_in_context` should care about this too?

Doesn't _really_ sound like **replacing** which is an issue the function already suffers from, because it also inserts it if source didn't exist. Maybe we're better of actually _ONLY_ replacing it and erroring otherwise so that it's clear when it fails to replace to begin with?

> NOTE: I don't think the function has to return bool.

I just did so that if you wanted to know whether it actually got replaced (the original removed and the new inserted at that same `instance` index) or was appended to the context. But since it's currently not used anywhere I can remove that. 👍

Reply:
Removed the `bool` return and made the function only replace and error otherwise.

Reply:
Used a different solution - completely removed the "replacing" of instances.